patches/minecraft/net/minecraft/server/level/ServerPlayer.java.patch

```diff
@@ -150,6 +150,15 @@
     }
  
     public boolean m_143432_() {
+@@ -1264,7 +_,7 @@
+ 
+    public void m_9213_(Entity p_9214_) {
+       Entity entity = this.m_8954_();
+-      this.f_8926_ = (Entity)(p_9214_ == null ? this : p_9214_);
++      this.f_8926_ = (Entity)(p_9214_ == null ? this : p_9214_ instanceof net.minecraftforge.entity.PartEntity ? ((net.minecraftforge.entity.PartEntity<?>) p_9214_).getParent() : p_9214_); // FORGE: fix MC-46486
```

Comment:
I wonder if this patch might be a bit cleaner by instead of doing this, on the next line adding:
```java
if (this.f_8926_ instanceof net.minecraftforge.entity.PartEntity<?> partEntity) this.f_8926_ = partEntity.getParent(); // FORGE: fix MC-46486
```
Though I forget if forge's patching system supports the pattern matching instanceof so if it doesn't then the extra cast would be needed, but either way I think it would slightly reduce the complexity of the patch and improve the readability of the code.